informer/informer.go

```diff
@@ -239,6 +240,14 @@ func (i *Informer) Watch(ctx context.Context) (chan watch.Event, chan watch.Even
 		close(errChan)
 	}()
 
+	// Before returning the event channels we wait for the cached being filled.
+	// The implication of an initial cache fill is a set up watcher connection to
+	// the remote Kubernetes watch API. It can happen that this setup takes longer
+	// due to retries after broken initial connections. Waiting for this setup to
+	// be properly done ensures users of the informer do not experience broken
+	// event processing due to initial connection issues.
+	<-i.filled
```

Comment:
This here plays together with the controller `Booted` magic. When calling `Informer.Watch` we did not always have a watcher successfully listening to the K8s API. This here ensures this.